ODIN_II/SRC/netlist_statistic.cpp

```diff
@@ -69,6 +69,42 @@ void init_stat(netlist_t* netlist) {
     netlist->num_logic_element = 0;
 }
 
+int calculate_multiplier_aware_critical_path(nnode_t* node, netlist_t* netlist) {
+    int i, j;
+    int result = 0;
+    if (node->traverse_visited != MULT_OPTIMIZATION_TRAVERSE_VALUE) {
+        /*this is a new node so depth visit it */
+        int backup_value = node->traverse_visited;
+        /* mark that we have visitied this node now */
+        node->traverse_visited = MULT_OPTIMIZATION_TRAVERSE_VALUE;
+        result = 1;
+        for (i = 0; i < node->num_output_pins; i++) {
```

Comment:
any chance you can just hook onto what is already there for the traversal?

Reply:
Thank you for the comment, I will consider linking with existing traversal subroutines =)